### PR TITLE
Allow users to generate a manifest for other platforms

### PR DESCRIPTION
**Pull request by jkeiser**
We are interested in generating manifests for chef and chef-dk tentatively, and checking them in to the repository. This patch allows us to do that from the command line for all OS's we support.

### CODE REVIEW

**Comment by schisamo** [comment]
@jkeiser Now that `omnibus manifest` has numerous CLI options we should add some Aruba coverage here:
https://github.com/chef/omnibus/tree/master/features/commands

**Comment by yzl** [comment]
:+1: pending Aruba coverage.

**Comment by jkeiser** [comment]
@yzl @schisamo I added aruba coverage. I had to modify `Omnibus.reset!` to reset cached project and software definitions to make it work, but it's in good shape now, I think. Also added a couple of build tests.

File: Gemfile [hunk 1]
```diff
@@ -3,6 +3,7 @@ gemspec
 
 # Fork to allow for a recent version of multipart-post.
 gem 'pedump', git: 'https://github.com/ksubrama/pedump', branch: 'patch-1'
+gem 'pry'
```
Comment (yzl) [inline Gemfile]: Do we want `pry` all the time, or is this for debugging?

File: features/step_definitions/generator_steps.rb [hunk 2]
```diff
@@ -28,6 +35,11 @@
   EOH
 end
 
+Given(/^I debug$/) do
```
Comment (yzl) [inline features/step_definitions/generator_steps.rb]: I'd rather not add this if it means adding `pry` as a dep, but will defer to others' opinion.

Comment (scotthain) [inline features/step_definitions/generator_steps.rb]: I'd also like to avoid `pry` as a dep if at all possible.

File: Gemfile (hunk 1 above)
Comment (jkeiser) [inline Gemfile]: This is for debugging only (thus putting in the gemfile so people don't get it when they `gem install omnibus`).

File: features/step_definitions/generator_steps.rb (hunk 2 above)
Comment (jkeiser) [inline features/step_definitions/generator_steps.rb]: `pry` isn't a dep, it's only in the `Gemfile`. `gem install omnibus` won't install it.

Comment (schisamo) [inline features/step_definitions/generator_steps.rb]: @jkeiser Omnibus isn't installed that way though...it's pulled in via a `bundle install` from an Omnibus project. Either way we should place `pry` in a `:development` group in the `Gemfile` OR probably just make it a development dependency in the gemspec:

```ruby
gem.add_development_dependency 'pry'
```

File: Gemfile (hunk 1 above)
Comment (sethvargo) [inline Gemfile]: It'll actually get pulled in unless you wrap it in `group :development`

File: features/step_definitions/generator_steps.rb (hunk 2 above)
Comment (sethvargo) [inline features/step_definitions/generator_steps.rb]: The advantage to putting it in the gemspec as a dev dep as opposed to in the gemfile is that it shows up on rubygems.org as a dev dep (gemfile deps dont, but still get pulled in)

File: Gemfile (hunk 1 above)
Comment (jkeiser) [inline Gemfile]: And in goes a `group :development`!

File: features/step_definitions/generator_steps.rb (hunk 2 above)
Comment (jkeiser) [inline features/step_definitions/generator_steps.rb]: Will do!

**Comment by jkeiser** [comment]
@schisamo @yzl @sethvargo development dependency fixed.

File: lib/omnibus/project.rb [hunk 3]
```diff
@@ -49,6 +49,10 @@ def load(name, manifest=nil)
         end
       end
 
+      def reset!
```
Comment (schisamo) [inline lib/omnibus/project.rb]: Need a yard comment here, see the following for a good example:
https://github.com/chef/omnibus/blob/798b2db8e34a296d074006086288cfb6693c9d65/lib/omnibus/config.rb#L76-L81

File: lib/omnibus/software.rb [hunk 4]
```diff
@@ -52,6 +52,10 @@ def load(project, name, manifest)
         end
       end
 
+      def reset!
```
Comment (schisamo) [inline lib/omnibus/software.rb]: Need a yard comment here, see the following for a good example:
https://github.com/chef/omnibus/blob/798b2db8e34a296d074006086288cfb6693c9d65/lib/omnibus/config.rb#L76-L81

File: lib/omnibus/project.rb (hunk 3 above)
Comment (jkeiser) [inline lib/omnibus/project.rb]: Done!

**Comment by schisamo** [comment]
@jkeiser Let's squash this down into 1 or 2 commits please (and ensure new feature commits have a descriptive message about the feature). Otherwise I am :+1:

**Comment by scotthain** [comment]
:+1: with squash

**Comment by jkeiser** [comment]
Squashed!

**Comment by schisamo** [comment]
@jkeiser MWR once Travis and AppVeyor checks are green.

**Pull request merged**